lib/collection/src/shards/replica_set/snapshots.rs

```diff
@@ -91,6 +114,84 @@ impl ShardReplicaSet {
             return Ok(false);
         }
 
+        let segments_path = LocalShard::segments_path(replica_path);
+
+        let mut snapshot_segments = HashSet::new();
+        let mut snapshot_manifests = SegmentManifests::default();
+
+        for segment_entry in segments_path.read_dir()? {
+            let segment_path = segment_entry?.path();
+
+            if !segment_path.is_dir() {
+                log::warn!(
+                    "segment path {} in extracted snapshot {} is not a directory",
+                    segment_path.display(),
+                    replica_path.display(),
+                );
+
+                continue;
+            }
```

Comment:
When can we expect this to happen?

Reply:
Never, but just in case? Debugging (even trivial) snapshots and file related bugs is *extremely* annoying, so I just check/log/contextualize everything.